ui/src/containers/VisualiseForm/StatementsForm/OptionsEditor.js

```diff
@@ -0,0 +1,154 @@
+      onChange={barChartGroupingLimitHandler}>
+      <option value={FIVE}>Show 5</option>
+      <option value={TEN}>Show 10</option>
+      <option value={FIFTEEN}>Show 15</option>
+      <option value={TWENTY}>Show 20</option>
+    </select>
+  </div>
+);
+
+const CounterEditorComponent = ({ model, benchmarkingHandler }) => (
+  <div className="form-group">
+    <label htmlFor="toggleInput">Enable Benchmarking</label>
+    <div id="toggleInput">
+      <Switch
+        id={'counterEditorComponent'}
+        checked={model.get('benchmarkingEnabled')}
+        onChange={benchmarkingHandler} />
+    </div>
+  </div>
+);
+
+export const XvsYOptionsEditor = compose(
+  connect(() => ({}), { updateModel }),
+  withHandlers({
+    trendLinesHandler: ({ updateModel: updateModelAction, model }) => (value) => {
+      updateModelAction({
+        schema: 'visualisation',
+        id: model.get('_id'),
+        path: 'trendLines',
+        value
+      });
+    },
+    sourceViewHandler: ({ updateModel: updateModelAction, model }) => {
+      updateModelAction({
+        schema: 'visualisation',
+        id: model.get('_id'),
+        path: 'sourceView',
+        value: !model.get('sourceView')
+      });
+    }
+  })
+)(XvsYOptionsEditorComponent);
+
+const BarEditor = compose(
+  connect(() => ({}), { updateModel, setInMetadata }),
+  withHandlers({
+    barChartGroupingLimitHandler: ({ updateModel: updateModelAction, model, setInMetadata: setInMetadataAction }) => (event) => {
+      updateModelAction({
+        schema: 'visualisation',
+        id: model.get('_id'),
+        path: 'barChartGroupingLimit',
+        value: parseInt(event.target.value)
+      });
+
+      setInMetadataAction({
+        schema: 'visualisation',
+        id: model.get('_id'),
+        path: ['activePage'],
+        value: 0
+      });
+    },
+    sourceViewHandler: ({ updateModel: updateModelAction, model }) => {
+      updateModelAction({
+        schema: 'visualisation',
+        id: model.get('_id'),
+        path: 'sourceView',
+        value: !model.get('sourceView')
+      });
+    }
+  })
+)(BarEditorComponent);
+
+const CounterEditor = compose(
+  connect(() => ({}), { updateModel }),
+  withHandlers({
+    benchmarkingHandler: ({ updateModel: updateModelAction, model }) => {
+      updateModelAction({
+        schema: 'visualisation',
+        id: model.get('_id'),
+        path: 'benchmarkingEnabled',
+        value: !model.get('benchmarkingEnabled')
+      });
+    }
+  })
+)(CounterEditorComponent);
+
+const DefaultEditor = compose(
+  connect(() => ({}), { updateModel }),
+  withHandlers({
+    sourceViewHandler: ({ updateModel: updateModelAction, model }) => {
+      updateModelAction({
+        schema: 'visualisation',
+        id: model.get('_id'),
+        path: 'sourceView',
+        value: !model.get('sourceView')
+      });
+    }
+  })
+)(DefaultEditorComponent);
+
+const OptionsEditor = ({ model }) => (
```

Comment:
Let's break these components out into their own files, but keep them contained in a folder together.

Reply:
Not sure then, perhaps it's already been broken out into separate files, maybe ask @cbishopvelti.